core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -73,7 +76,9 @@ public void loadNext() throws IOException {
     numRecordsRemaining--;
     if (numRecordsRemaining == 0) {
       in.close();
-      file.delete();
+      if (!file.delete()) {
+        logger.error("Unable to delete spill file {}", file.getPath());
```

Comment:
I did some search prior to making the change:

$ find . -name '*.java' -exec grep 'file.delete(' {} \; -print
      if (spill.file.exists() && !spill.file.delete()) {
./core/src/main/java/org/apache/spark/shuffle/unsafe/UnsafeShuffleExternalSorter.java
        if (spill.file.exists() && ! spill.file.delete()) {
./core/src/main/java/org/apache/spark/shuffle/unsafe/UnsafeShuffleWriter.java
        if (!file.delete()) {
./core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java
      if (!file.delete()) {
./core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java
    boolean deleted = file.delete();
./network/common/src/main/java/org/apache/spark/network/util/JavaUtils.java

This file was the only one among the files where delete() is called but whose return value wasn't checked.